tests/unit/notification/NotificationConfigManager.js

```diff
@@ -0,0 +1,178 @@
+                },
+            ],
+        },
+    };
+    return data;
+}
+
+function populateTestConfigs(zkClient, numberOfConfigs, cb) {
+    async.timesLimit(numberOfConfigs, concurrency, (n, done) => {
+        const bucket = `${bucketPrefix}${n + 1}`;
+        const val = getTestConfigValue(bucket);
+        const strVal = JSON.stringify(val);
+        const node = `/${zkConfigParentNode}/${bucket}`;
+        async.series([
+            nxt => zkClient.mkdirp(node, nxt),
+            nxt => zkClient.setData(node, strVal, nxt),
+        ], done);
+    }, cb);
+}
+
+function listBuckets(zkClient, cb) {
+    const node = `/${zkConfigParentNode}`;
+    zkClient.getChildren(node, cb);
+}
+
+function deleteTestConfigs(zkClient, cb) {
+    const node = `/${zkConfigParentNode}`;
+    listBuckets(zkClient, (e, children) => {
+        if (e) {
+            assert.ifError(e);
+            cb(e);
+        } else {
+            async.eachLimit(children, concurrency, (child, next) => {
+                const childNode = `${node}/${child}`;
+                zkClient.remove(childNode, next);
+            }, cb);
+        }
+    });
+}
+
+describe('NotificationConfigManager', () => {
+    const zkClient = new ZookeeperMock();
+    const params = {
+        zkClient,
+        parentNode: zkConfigParentNode,
+        logger,
+    };
+
+    function checkCount(manager, cb) {
+        listBuckets(zkClient, (err, buckets) => {
+            assert.ifError(err);
+            const zkConfigCount = buckets.length;
+            const managerConfigs = manager.getBucketsWithConfigs();
+            assert.strictEqual(zkConfigCount, managerConfigs.length);
+            cb();
+        });
+    }
+
+    function managerInit(manager, cb) {
+        manager.init(err => {
+            assert.ifError(err);
+            cb();
+        });
+    }
+
+    beforeEach(done => populateTestConfigs(zkClient, 5, done));
+
+    afterEach(() => {
+        zkClient._resetState();
+    });
+
+    it('constructor and init checks', done => {
+        assert.throws(() => new NotificationConfigManager());
+        assert.throws(() => new NotificationConfigManager({}));
+        assert.throws(() => new NotificationConfigManager({
+            zkClient: null,
+            logger: null,
+        }));
+        const manager = new NotificationConfigManager(params);
+        assert(manager instanceof NotificationConfigManager);
+        async.series([
+            next => managerInit(manager, next),
+            next => checkCount(manager, next),
+        ], done);
+    });
+
+    it('should get bucket notification configuration', () => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket1';
+            const result = manager.getConfig(bucket);
+            assert.strictEqual(result.bucket, bucket);
+        });
+    });
+
+    it('should return undefined for an invalid bucket', () => {
```

Comment:
the description might say `for a bucket that does not have a notification config` or that the bucket itself does not exist, or something similar? Since the bucket name does not look invalid it looks confusing.